translation/src/main/java/com/google/cloud/trace/zipkin/translation/TraceTranslator.java

```diff
@@ -89,15 +89,13 @@ private void finishTrace(Trace.Builder traceBuilder, Collection<Trace> converted
     }
   }
 
-  private String convertTraceId(long zipkinTraceId) {
-    // Stackdriver trace ID's are 128 bits = 16 bytes * 8
-    ByteBuffer idBuffer = ByteBuffer.allocate(16);
-    idBuffer.putLong(zipkinTraceId);
-    idBuffer.putLong(zipkinTraceId);
-    StringBuilder idBuilder = new StringBuilder();
-    for (byte b : idBuffer.array()) {
-      idBuilder.append(String.format("%02x", b));
-    }
-    return idBuilder.toString();
+  // Stackdriver trace ID's are 128 bits = 16 bytes * 8
+  private String convertTraceId(Span zipkinSpan) {
+    // zipkin conditionally encodes its id as 16 chars if the high bits are zero
+    // This unconditionally encodes as 32 chars as Stackdriver trace IDs are fixed length.
+    char[] result = new char[32];
+    Util.writeHexLong(result, 0, zipkinSpan.traceIdHigh);
```

Comment:
I'm OK with this if you're OK with this depending on zipkin.internal.Util.

We may want to have this write `zipkinSpan.traceId` for both the high and low bits if `zipkinSpan.traceIdHigh` isn't set. Some users call our API and order results by traceId to get a "random sample" of traces. Duplicating the 64-bit ID instead of padding the ID with zeros will remove some sampling bias for any projects that contain both Zipkin and Stackdriver traces. Do you think that would cause any problems?